govcd/catalog_test.go

```diff
@@ -29,5 +29,44 @@ func (vcd *TestVCD) Test_FindCatalogItem(check *C) {
 	}
 	// Test non-existant catalog item
 	catitem, err = cat.FindCatalogItem("INVALID")
-	check.Assert(err, NotNil)
+	check.Assert(catitem, Equals, CatalogItem{})
+	check.Assert(err, IsNil)
+}
+
+// Creates a Catalog, updates the description, and checks the changes against the
+// newly updated catalog. Then deletes the catalog
+func (vcd *TestVCD) Test_UpdateCatalog(check *C) {
+	org, err := GetAdminOrgByName(vcd.client, vcd.config.VCD.Org)
+	check.Assert(org, Not(Equals), AdminOrg{})
+	check.Assert(err, IsNil)
+	adminCatalog, err := org.CreateCatalog("UpdateCatalogTest", "UpdateCatalogTest", true)
```

Comment:
The procedure should check whether the catalog already exists, and then either delete it or skip the creation.
Also, and _this request extends to all CRUD tests,_ the literal strings that are repeated (`"UpdateCatalogTest"`, `"DeleteCatalogTest"`, `"CatalogCreationTest"`) should become a local variable.